upup/pkg/fi/cloudup/gce/gce_cloud.go

```diff
@@ -29,24 +30,41 @@ import (
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns"
 )
 
-type GCECloud struct {
-	Compute *compute.Service
-	Storage *storage.Service
+type GCECloud interface {
```

Comment:
I think the better abstraction is to go a level higher and put both GCE cloud and AWS cloud under the same interface instead of assign both different ones. The point of the interfaces here would be to avoid having to do `switch` statements. I know this is a LOT of work, but I wonder if its the right direction going forward